app/controllers/admin/application_pets_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Admin::ApplicationPetsController < ApplicationController
+
+  def update
```

Comment:
It's okay to keep this controller action in your other non-admin ApplicationPets controller if you want! The route from the button to approve a pet doesn't need to be prefixed with `/admin` so it could go in the regular ApplicationPets controller. But this is not "wrong" ! Just another option